staging/src/k8s.io/apiserver/pkg/server/dynamiccertificates/cert_key.go

```diff
@@ -28,16 +28,35 @@ type CertKeyContentProvider interface {
 	CurrentCertKeyContent() ([]byte, []byte)
 }
 
+// CertKeyContentProvider provides a certificate and matching private key
+type SNICertKeyContentProvider interface {
+	CertKeyContentProvider
+	// SNINames provides names used for SNI. May return nil.
+	SNINames() []string
+}
+
 // caBundleContent holds the content for the cert and key
 type certKeyContent struct {
 	cert []byte
 	key  []byte
+
+	sniNames []string
```

Comment:
sorry, thought it was implied from the above comment. This should be in a distinct struct too so we never confuse which is which.  The meanings are slightly different.